src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BitcoinHeaderVersionRule.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Networks;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Checks if <see cref="BitcoinMain"/> network block's header has a valid block version.
+    /// <seealso cref="BitcoinActivationRule" />
+    /// </summary>
+    [HeaderValidationRule(CanSkipValidation = true)]
+    public class BitcoinHeaderVersionRule : HeaderVersionRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.BadVersion">Thrown if block's version is outdated or otherwise invalid.</exception>
+        public override Task RunAsync(RuleContext context)
+        {
+            Guard.NotNull(context.ConsensusTip, nameof(context.ConsensusTip));
+
+            ChainedHeader chainedHeader = context.ValidationContext.ChainedHeader;
+
+            // BIP9 mandates that the top bits of version be 001. So a standard node should never generate
+            // block versions between 4 and 0x20000000. Block versions 5 onwards were never allocated, as the
+            // BIP9 standard became predominant.
+            if ((chainedHeader.Header.Version > 4) && (chainedHeader.Header.Version < ThresholdConditionCache.VersionbitsTopBits))
```

Comment:
BIP9 voting requires a version of >= ThresholdConditionCache.VersionbitsTopBits, unless I am very much mistaken.

Reply:
https://www.blockchain.com/btc/block-height/455757

Reply:
That is a good find, and reason enough to change the check, but note that it is not BIP9 voting.

Shouldn't the header sync on BTC mainnet have caught this? Does the 'assume valid' functionality bypass this rule?

Reply:
CanSkipValidation = true, Which means that you need to disable assume valid or put it below this block if you want to test it because it would be skipped. You also need to disable checkpoints or delete checkpoints that are over this block number

Reply:
The master branch still uses the old way of syncing, so the header validation would not execute this rule